wasm/src/program/transaction.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the Aleo library.
+
+// The Aleo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Aleo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Aleo library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{
+    account::{Address, PrivateKey},
+    record::RecordPlaintext,
+};
+
+use aleo_account::{Aleo, Process, Program, Request, Transaction, PARAMETER_PROVIDER};
+use std::convert::TryInto;
+use wasm_bindgen::prelude::*;
+use js_sys::Array;
+
+#[wasm_bindgen]
+pub struct ParameterProvider {}
+
+#[wasm_bindgen]
+impl ParameterProvider {
```

Comment:
This is the core interface exposed to the client. A client now becomes responsible for handling files and then injecting them here. We could probably make this nicer to handle the credits program.